core/src/main/java/jenkins/cli/StopBuildsCommand.java

```diff
@@ -63,63 +64,55 @@ protected int run() throws Exception {
             if (item instanceof Job) {
                 jobsToStop.add((Job) item);
             } else {
-                throw new IllegalArgumentException("Invalid job name = " + jobName);
+                throw new IllegalArgumentException("Job not found: '" + jobName + "'");
             }
         }
 
         for (final Job job : jobsToStop) {
-            stopJobBuilds(job, job.getName());
+            stopJobBuilds(job);
+        }
+
+        if (!isAnyBuildStopped) {
+            stdout.println("No builds stopped");
         }
 
         return 0;
     }
 
-    private void stopJobBuilds(final Job job,
-                               final String jobName) throws IOException, ServletException {
+    private void stopJobBuilds(final Job job) {
         final Run lastBuild = job.getLastBuild();
-        final List<String> stoppedBuildsNames = new ArrayList<>();
+        final String jobName = job.getFullDisplayName();
         if (lastBuild != null && lastBuild.isBuilding()) {
-            stopBuild(lastBuild, jobName, stoppedBuildsNames);
-            checkAndStopPreviousBuilds(lastBuild, jobName, stoppedBuildsNames);
+            stopBuild(lastBuild, jobName);
+            checkAndStopPreviousBuilds(lastBuild, jobName);
         }
-        updateResultOutput(jobName, stoppedBuildsNames);
     }
 
     private void stopBuild(final Run build,
-                           final String jobName,
-                           final List<String> stoppedBuildNames) throws IOException, ServletException {
+                           final String jobName) {
         final String buildName = build.getDisplayName();
-        stoppedBuildNames.add(buildName);
         Executor executor = build.getExecutor();
         if (executor != null) {
-            executor.doStop();
+            try {
+                executor.doStop();
+                isAnyBuildStopped = true;
+                stdout.println(String.format("Build '%s' stopped for job '%s'", buildName, jobName));
+            } catch (final Exception e) {
+                stdout.print(String.format("Exception occurred while trying to stop build '%s' for job '%s'. ", buildName, jobName));
```

Comment:
Better to delete this `try`-`catch` block and let any exception be thrown up out of the command. The only _expected_ exception would be `AccessDeniedException`, which is better handled by `CLICommand` with a standardized exit code and everything. You do not want the command succeeding (exit code 0) if in fact an error occurred here.

Reply:
[Here](https://github.com/jenkinsci/jenkins/pull/3686#discussion_r277089004) was a proposal to handle `AccessDeniedException`.
And [here](https://github.com/jenkinsci/jenkins/pull/3686#discussion_r278850305) another proposal to continue stopping even if some operation failed. 
I agree that exit code 0 is wrong for situation where some build stopping failed.

Here is suggestion - handle here only 'AccessDeniedException', log every such exception and rethrow last one after attempt to stop builds for every jobs from input arguments. 
What do you think, @jglick ?